src/Kiota.Builder/CodeDOM/CodeClass.cs

```diff
@@ -154,6 +252,11 @@ public class ClassDeclaration : ProprietableBlockDeclaration
     {
         get => inherits; set
         {
+            if (value != null && !value.IsExternal && Parent is CodeClass codeClass && codeClass.Properties.Any())
+            {
+                throw new InvalidOperationException("Cannot change the inherits-property of an already populated type");
```

Comment:
@Ndiritu you are right, those changes caused some issues with this PR, which is why this exception was added. This PR also updates the tests to make sure base types are not added to classes where properties have already been added.

Reply:
I was in a bit of a rush to get 1.4.0 out and compromised the best I could to avoid negatively impacting the Graph SDKs/existing clients out there on the stable languages. Sorry about that. For the same reasons we're trying to address in this PR, the inheritance structure should be built from base to derived, not the other way around.

Reply:
No worries! Thanks for clarifying.